code/game/mecha/equipment/mecha_equipment.dm

```diff
@@ -15,6 +15,7 @@
 	var/salvageable = 1
 	var/selectable = 1	// Set to 0 for passive equipment such as mining scanner or armor plates
 	var/harmful = FALSE //Controls if equipment can be used to attack by a pacifist.
+	var/list/mecha_types = list(/obj/mecha) //Mecha types this equipment can be attached to

```

Comment:
If you're not removing the  `can_attach` proc I don't see why this is beneficial.

Instead of putting it behind istypes in a proc that fires situationally you're now generating a list per mech item created.

Reply:
Also undocumented which is yucky.